### DISCUSSION TITLE
Suggestion: Allow the "options" field property to be a function similar to hooks

### DISCUSSION THREAD

**Discussion by hades200082** [payloadcms/payload#1899] [Feature Requests & Ideas]
As mentioned in Discord ( https://discord.com/channels/967097582721572934/1064471608833609728 )

My idea here is that, for example with "select" fields, if the "options" property were to accept a function, and that function could access the data of other fields in the current collection/global, then it would become possible to have fields whose available options are dynamic based on the other values in the content item.

My thoughts on this so far are that it might also need some additional properties to reduce the amount of re-rendering going on. Example definition below:

```javascript
import { CollectionConfig } from 'payload/types';

const Orders: CollectionConfig = {
	slug: 'orders',
	fields: [
		{
			name: 'productType',
			type: 'select',
			required: true,
			options: [
				'One',
				'Two'
				'Three'
			],
			hasDependingFields: true, // this tells Payload that when this field's value changes it should look for other fields that depend on this one and re-render them
		},
		{
			name: 'productSubType',
			type: 'select',
			required: true,
			dependsOnFields: [ 'productType' ]
			options: ({data}) => { // To be safer here this could be passed an object containing only the fields listed in `dependsOnFields` above
				if(data.productType === 'One') {
					return ['One A', 'One B'];
				}
				if(data.productType === 'Two') {
					return ['Two A', 'Two B'];
				}
				if(data.productType === 'Three') {
					return ['Three A', 'Three B'];
				}
			}
		}
	]
};
```

I've made a couple of assumptions here:

1. Payload is capable of re-rendering a field control in the admin based on changes in another field. This seems to already be possible with the conditional fields feature.
3. The re-rendering of a field would be able to call the passed function again

**Comment by ssyberg** [comment]
Is there an architectural reason why this can't be done?

**Comment by ssyberg** [comment]
Ah pulling this from https://github.com/payloadcms/payload/discussions/564#discussioncomment-2753836

From @DanRibbens 
> Currently, no. The setup for graphql and database schema, and validation are all strict around the predefined values.
